CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,613 @@
+
+    // Update feature types for the newly selected table
+    GeodatabaseFeatureTable *table = m_tablesByName.value(tableName, nullptr);
+    if (table)
+    {
+      updateFeatureTypesForTable(table);
+    }
+  }
+}
+
+void EditGeodatabaseWithTransactions::addFeatureAtLocation(const QString &tableName, const QString &featureTypeName, const QPoint &location)
+{
+  if (!m_mapView || !m_geodatabase)
+  {
+    return;
+  }
+
+  // Get the correct table using dynamic lookup
+  GeodatabaseFeatureTable *targetTable = m_tablesByName.value(tableName, nullptr);
+
+  if (!targetTable)
+  {
+    setMessageText(QString("Table '%1' not found.").arg(tableName));
+    return;
+  }
+
+  // Find the feature type
+  FeatureType selectedFeatureType;
+  bool foundType = false;
+  for (const FeatureType &featureType : targetTable->featureTypes())
+  {
+    if (featureType.name() == featureTypeName)
+    {
+      selectedFeatureType = featureType;
+      foundType = true;
+      break;
+    }
+  }
+
+  if (!foundType)
+  {
+    setMessageText(QString("Feature type '%1' not found in table '%2'.").arg(featureTypeName, tableName));
+    return;
+  }
+
+  // Convert screen point to map point
+  Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
+
+  if (mapPoint.isEmpty())
+  {
+    setMessageText("Invalid location clicked.");
+    return;
+  }
+
+  // Convert point to WGS84 for comparison with extent (extent is in WGS84)
+  Point wgs84Point = mapPoint;
+  if (!mapPoint.spatialReference().isEmpty() && mapPoint.spatialReference().wkid() != 4326)
+  {
+    // Project the point to WGS84
+    Geometry projectedGeometry = GeometryEngine::project(mapPoint, SpatialReference::wgs84());
+    if (!projectedGeometry.isEmpty())
+    {
+      wgs84Point = static_cast<Point>(projectedGeometry);
+    }
+  }
+
+  // Check if point is within extent
+  bool withinBounds = (wgs84Point.x() >= m_extent.xMin() && wgs84Point.x() <= m_extent.xMax()
+                       && wgs84Point.y() >= m_extent.yMin()
+                       && wgs84Point.y() <= m_extent.yMax());
+
+  if (!withinBounds)
+  {
+    setMessageText("Error: Feature geometry is outside the generate geodatabase geometry.");
+    return;
+  }
+
+  // Create feature using the selected type's template attributes
+  QVariantMap attributes;
+  if (!selectedFeatureType.templates().empty())
+  {
+    // Use the first template's attributes if available
+    attributes = selectedFeatureType.templates().at(0).prototypeAttributes();
+  }
+
+  Feature *newFeature = targetTable->createFeature(attributes, mapPoint);
+  if (!newFeature)
+  {
+    setMessageText("Failed to create new feature.");
+    return;
+  }
+
+  // Add the new feature to the table
+  QFuture<void> addFuture = targetTable->addFeatureAsync(newFeature);
+  QFutureWatcher<void> *addWatcher = new QFutureWatcher<void>(this);
+
+  connect(addWatcher, &QFutureWatcher<void>::finished, this, [this, addWatcher, targetTable]()
+  {
+    try
+    {
```

Comment:
I get the sense that you didn't make many of the changes I recommended. I won't review any further at this point since I suspect that was a mistake. I could be wrong. Let me know.